{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import olga.load_model as load_model\n",
    "import olga.generation_probability as pgen\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('../data/output_data/exp_AB_wide4.csv', index_col=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>junction_aa</th>\n",
       "      <th>junction</th>\n",
       "      <th>v_call</th>\n",
       "      <th>j_call</th>\n",
       "      <th>decombinator_id</th>\n",
       "      <th>ID</th>\n",
       "      <th>control</th>\n",
       "      <th>PCR</th>\n",
       "      <th>chain</th>\n",
       "      <th>proportion_-3</th>\n",
       "      <th>...</th>\n",
       "      <th>total_-2</th>\n",
       "      <th>total_-1</th>\n",
       "      <th>total_0</th>\n",
       "      <th>total_1</th>\n",
       "      <th>total_2</th>\n",
       "      <th>total_3</th>\n",
       "      <th>total_4</th>\n",
       "      <th>total_14</th>\n",
       "      <th>max_timepoint</th>\n",
       "      <th>max_timepoint_class</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CAARAIGNTPLVF</td>\n",
       "      <td>TGTGCAGCAAGGGCCATCGGAAACACACCTCTTGTCTTT</td>\n",
       "      <td>TRAV29/DV5</td>\n",
       "      <td>TRAJ29</td>\n",
       "      <td>23, 17, 2, 8, GGCCATC</td>\n",
       "      <td>17</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>alpha</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>85012.0</td>\n",
       "      <td>90223.0</td>\n",
       "      <td>78873.0</td>\n",
       "      <td>65517.0</td>\n",
       "      <td>65456.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>proportion_1</td>\n",
       "      <td>und</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CAASIWNYGGSQGNLIF</td>\n",
       "      <td>TGTGCAGCAAGTATATGGAATTATGGAGGAAGCCAAGGAAATCTCA...</td>\n",
       "      <td>TRAV13-1</td>\n",
       "      <td>TRAJ42</td>\n",
       "      <td>6, 31, 0, 1, TATG</td>\n",
       "      <td>17</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>alpha</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>85012.0</td>\n",
       "      <td>90223.0</td>\n",
       "      <td>78873.0</td>\n",
       "      <td>65517.0</td>\n",
       "      <td>65456.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>proportion_2</td>\n",
       "      <td>und</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CAASNQAGTALIF</td>\n",
       "      <td>TGTGCAGCAAGTAACCAGGCAGGAACTGCTCTGATCTTT</td>\n",
       "      <td>TRAV13-1</td>\n",
       "      <td>TRAJ15</td>\n",
       "      <td>6, 5, 0, 3,</td>\n",
       "      <td>17</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>alpha</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>85012.0</td>\n",
       "      <td>90223.0</td>\n",
       "      <td>78873.0</td>\n",
       "      <td>65517.0</td>\n",
       "      <td>65456.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>proportion_2</td>\n",
       "      <td>und</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CACASKIIF</td>\n",
       "      <td>TGTGCTTGTGCTTCCAAGATAATCTTT</td>\n",
       "      <td>TRAV24</td>\n",
       "      <td>TRAJ3</td>\n",
       "      <td>18, 18, 5, 11, TT</td>\n",
       "      <td>17</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>alpha</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>85012.0</td>\n",
       "      <td>90223.0</td>\n",
       "      <td>78873.0</td>\n",
       "      <td>65517.0</td>\n",
       "      <td>65456.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>proportion_2</td>\n",
       "      <td>und</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>CACGNTGFQKLVF</td>\n",
       "      <td>TGTGCTTGCGGGAACACAGGCTTTCAGAAACTTGTATTT</td>\n",
       "      <td>TRAV2</td>\n",
       "      <td>TRAJ8</td>\n",
       "      <td>13, 48, 8, 1, TGCGG</td>\n",
       "      <td>17</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>alpha</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>85012.0</td>\n",
       "      <td>90223.0</td>\n",
       "      <td>78873.0</td>\n",
       "      <td>65517.0</td>\n",
       "      <td>65456.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>proportion_1</td>\n",
       "      <td>und</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         junction_aa                                           junction  \\\n",
       "1      CAARAIGNTPLVF            TGTGCAGCAAGGGCCATCGGAAACACACCTCTTGTCTTT   \n",
       "2  CAASIWNYGGSQGNLIF  TGTGCAGCAAGTATATGGAATTATGGAGGAAGCCAAGGAAATCTCA...   \n",
       "3      CAASNQAGTALIF            TGTGCAGCAAGTAACCAGGCAGGAACTGCTCTGATCTTT   \n",
       "4          CACASKIIF                        TGTGCTTGTGCTTCCAAGATAATCTTT   \n",
       "5      CACGNTGFQKLVF            TGTGCTTGCGGGAACACAGGCTTTCAGAAACTTGTATTT   \n",
       "\n",
       "       v_call  j_call        decombinator_id  ID  control  PCR  chain  \\\n",
       "1  TRAV29/DV5  TRAJ29  23, 17, 2, 8, GGCCATC  17     True  NaN  alpha   \n",
       "2    TRAV13-1  TRAJ42      6, 31, 0, 1, TATG  17     True  NaN  alpha   \n",
       "3    TRAV13-1  TRAJ15           6, 5, 0, 3,   17     True  NaN  alpha   \n",
       "4      TRAV24   TRAJ3      18, 18, 5, 11, TT  17     True  NaN  alpha   \n",
       "5       TRAV2   TRAJ8    13, 48, 8, 1, TGCGG  17     True  NaN  alpha   \n",
       "\n",
       "   proportion_-3  ...  total_-2  total_-1  total_0  total_1  total_2  total_3  \\\n",
       "1            NaN  ...       NaN       NaN  85012.0  90223.0  78873.0  65517.0   \n",
       "2            NaN  ...       NaN       NaN  85012.0  90223.0  78873.0  65517.0   \n",
       "3            NaN  ...       NaN       NaN  85012.0  90223.0  78873.0  65517.0   \n",
       "4            NaN  ...       NaN       NaN  85012.0  90223.0  78873.0  65517.0   \n",
       "5            NaN  ...       NaN       NaN  85012.0  90223.0  78873.0  65517.0   \n",
       "\n",
       "   total_4  total_14  max_timepoint  max_timepoint_class  \n",
       "1  65456.0       NaN   proportion_1                  und  \n",
       "2  65456.0       NaN   proportion_2                  und  \n",
       "3  65456.0       NaN   proportion_2                  und  \n",
       "4  65456.0       NaN   proportion_2                  und  \n",
       "5  65456.0       NaN   proportion_1                  und  \n",
       "\n",
       "[5 rows x 29 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "generative_model = load_model.GenerativeModelVDJ()\n",
    "marginals_file_name = '../../OLGA/olga/default_models/human_T_beta/model_marginals.txt'\n",
    "generative_model.load_and_process_igor_model(marginals_file_name)\n",
    "\n",
    "params_file_name = '../../OLGA/olga/default_models/human_T_beta/model_params.txt'\n",
    "V_anchor_pos_file ='../../OLGA/olga/default_models/human_T_beta/V_gene_CDR3_anchors.csv'\n",
    "J_anchor_pos_file = '../../OLGA/olga/default_models/human_T_beta/J_gene_CDR3_anchors.csv'\n",
    "genomic_data = load_model.GenomicDataVDJ()\n",
    "genomic_data.load_igor_genomic_data(params_file_name, V_anchor_pos_file, J_anchor_pos_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "pgen_b = pgen.GenerationProbabilityVDJ(generative_model, genomic_data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "generative_model = load_model.GenerativeModelVJ()\n",
    "marginals_file_name = '../../OLGA/olga/default_models/human_T_alpha/model_marginals.txt'\n",
    "generative_model.load_and_process_igor_model(marginals_file_name)\n",
    "\n",
    "params_file_name = '../../OLGA/olga/default_models/human_T_alpha/model_params.txt'\n",
    "V_anchor_pos_file ='../../OLGA/olga/default_models/human_T_alpha/V_gene_CDR3_anchors.csv'\n",
    "J_anchor_pos_file = '../../OLGA/olga/default_models/human_T_alpha/J_gene_CDR3_anchors.csv'\n",
    "genomic_data = load_model.GenomicDataVJ()\n",
    "genomic_data.load_igor_genomic_data(params_file_name, V_anchor_pos_file, J_anchor_pos_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "pgen_a = pgen.GenerationProbabilityVJ(generative_model, genomic_data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "            junction_aa                                           junction  \\\n",
      "1         CAARAIGNTPLVF            TGTGCAGCAAGGGCCATCGGAAACACACCTCTTGTCTTT   \n",
      "2     CAASIWNYGGSQGNLIF  TGTGCAGCAAGTATATGGAATTATGGAGGAAGCCAAGGAAATCTCA...   \n",
      "3         CAASNQAGTALIF            TGTGCAGCAAGTAACCAGGCAGGAACTGCTCTGATCTTT   \n",
      "4             CACASKIIF                        TGTGCTTGTGCTTCCAAGATAATCTTT   \n",
      "5         CACGNTGFQKLVF            TGTGCTTGCGGGAACACAGGCTTTCAGAAACTTGTATTT   \n",
      "...                 ...                                                ...   \n",
      "9529     CASSPGPKYNEQFF         TGTGCCAGCAGCCCGGGACCGAAGTACAATGAGCAGTTCTTC   \n",
      "9530    CASSQDVLAKNIQYF      TGTGCCAGCAGCCAAGACGTCCTAGCCAAAAACATTCAGTACTTC   \n",
      "9531     CASSQEEGSSEAFF         TGTGCCAGCAGCCAAGAGGAGGGATCCTCTGAAGCTTTCTTT   \n",
      "9532    CASSQVTLSNAEAFF      TGTGCCAGCAGCCAAGTTACGTTGAGCAACGCTGAAGCTTTCTTT   \n",
      "9533  CASSTPNGLAAGDTQYF  TGTGCCAGCTCGACCCCAAATGGACTAGCGGCGGGAGATACGCAGT...   \n",
      "\n",
      "          v_call   j_call                         decombinator_id   ID  \\\n",
      "1     TRAV29/DV5   TRAJ29                   23, 17, 2, 8, GGCCATC   17   \n",
      "2       TRAV13-1   TRAJ42                       6, 31, 0, 1, TATG   17   \n",
      "3       TRAV13-1   TRAJ15                            6, 5, 0, 3,    17   \n",
      "4         TRAV24    TRAJ3                       18, 18, 5, 11, TT   17   \n",
      "5          TRAV2    TRAJ8                     13, 48, 8, 1, TGCGG   17   \n",
      "...          ...      ...                                     ...  ...   \n",
      "9529      TRBV27  TRBJ2-1              18, 6, 6, 4, CCCGGGACCGAAG  373   \n",
      "9530     TRBV4-2  TRBJ2-4                     24, 9, 0, 0, CGTCCT  373   \n",
      "9531     TRBV3-1  TRBJ1-1                21, 0, 0, 6, GGAGGGATCCT  373   \n",
      "9532     TRBV3-1  TRBJ1-1            21, 0, 1, 6, TTACGTTGAGCAACG  373   \n",
      "9533    TRBV12-4  TRBJ2-3  6, 8, 8, 5, TCGACCCCAAATGGACTAGCGGCGGG  373   \n",
      "\n",
      "      control  PCR  chain  proportion_-3  ...  total_-2  total_-1   total_0  \\\n",
      "1        True  NaN  alpha            NaN  ...       NaN       NaN   85012.0   \n",
      "2        True  NaN  alpha            NaN  ...       NaN       NaN   85012.0   \n",
      "3        True  NaN  alpha            NaN  ...       NaN       NaN   85012.0   \n",
      "4        True  NaN  alpha            NaN  ...       NaN       NaN   85012.0   \n",
      "5        True  NaN  alpha            NaN  ...       NaN       NaN   85012.0   \n",
      "...       ...  ...    ...            ...  ...       ...       ...       ...   \n",
      "9529    False  0.0   beta            NaN  ...       NaN       NaN  145866.0   \n",
      "9530    False  0.0   beta            NaN  ...       NaN       NaN  145866.0   \n",
      "9531    False  0.0   beta            NaN  ...       NaN       NaN  145866.0   \n",
      "9532    False  0.0   beta            NaN  ...       NaN       NaN  145866.0   \n",
      "9533    False  0.0   beta            NaN  ...       NaN       NaN  145866.0   \n",
      "\n",
      "       total_1  total_2   total_3  total_4  total_14  max_timepoint  \\\n",
      "1      90223.0  78873.0   65517.0  65456.0       NaN   proportion_1   \n",
      "2      90223.0  78873.0   65517.0  65456.0       NaN   proportion_2   \n",
      "3      90223.0  78873.0   65517.0  65456.0       NaN   proportion_2   \n",
      "4      90223.0  78873.0   65517.0  65456.0       NaN   proportion_2   \n",
      "5      90223.0  78873.0   65517.0  65456.0       NaN   proportion_1   \n",
      "...        ...      ...       ...      ...       ...            ...   \n",
      "9529  112938.0      NaN  200937.0      NaN  111327.0            NaN   \n",
      "9530  112938.0      NaN  200937.0      NaN  111327.0            NaN   \n",
      "9531  112938.0      NaN  200937.0      NaN  111327.0            NaN   \n",
      "9532  112938.0      NaN  200937.0      NaN  111327.0            NaN   \n",
      "9533  112938.0      NaN  200937.0      NaN  111327.0            NaN   \n",
      "\n",
      "      max_timepoint_class  \n",
      "1                     und  \n",
      "2                     und  \n",
      "3                     und  \n",
      "4                     und  \n",
      "5                     und  \n",
      "...                   ...  \n",
      "9529                  und  \n",
      "9530                  und  \n",
      "9531                  und  \n",
      "9532                  und  \n",
      "9533                  und  \n",
      "\n",
      "[9533 rows x 29 columns]\n"
     ]
    }
   ],
   "source": [
    "print(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "df['pgen'] = ''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "dfa = df.loc[df['chain'] == 'alpha']\n",
    "dfb = df.loc[df['chain'] == 'beta']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1497382/1919900756.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  dfa['pgen'] = dfa.apply(lambda row:\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Unfamiliar V gene/allele: TRBV21-1\n",
      "No recognized V genes/alleles. Using default V_usage_mask\n",
      "Unfamiliar V gene/allele: TRBV21-1\n",
      "No recognized V genes/alleles. Using default V_usage_mask\n",
      "Unfamiliar V gene/allele: TRBV21-1\n",
      "No recognized V genes/alleles. Using default V_usage_mask\n",
      "Unfamiliar V gene/allele: TRBV21-1\n",
      "No recognized V genes/alleles. Using default V_usage_mask\n",
      "Unfamiliar V gene/allele: TRBV21-1\n",
      "No recognized V genes/alleles. Using default V_usage_mask\n",
      "Unfamiliar V gene/allele: TRBV21-1\n",
      "No recognized V genes/alleles. Using default V_usage_mask\n",
      "Unfamiliar V gene/allele: TRBV21-1\n",
      "No recognized V genes/alleles. Using default V_usage_mask\n",
      "Unfamiliar V gene/allele: TRBV21-1\n",
      "No recognized V genes/alleles. Using default V_usage_mask\n",
      "Unfamiliar V gene/allele: TRBV21-1\n",
      "No recognized V genes/alleles. Using default V_usage_mask\n",
      "Unfamiliar V gene/allele: TRBV21-1\n",
      "No recognized V genes/alleles. Using default V_usage_mask\n",
      "Unfamiliar V gene/allele: TRBV21-1\n",
      "No recognized V genes/alleles. Using default V_usage_mask\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1497382/1919900756.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  dfb['pgen'] = dfb.apply(lambda row:\n"
     ]
    }
   ],
   "source": [
    "dfa['pgen'] = dfa.apply(lambda row: \n",
    "                        pgen_a.compute_aa_CDR3_pgen(row['junction_aa'], \n",
    "                                row['v_call'], print_warnings= True), axis=1)\n",
    "dfb['pgen'] = dfb.apply(lambda row: \n",
    "                        pgen_b.compute_aa_CDR3_pgen(row['junction_aa'], \n",
    "                        row['v_call'], print_warnings= True), axis=1)\n",
    "# TRBV1, TRBV17, TRBV21-1, TRBV23-1, TRBV5-3, TRBV5-7, TRBV6-7, TRBV7-5 are pseudogenes that decombinator does not clean up"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>junction_aa</th>\n",
       "      <th>junction</th>\n",
       "      <th>v_call</th>\n",
       "      <th>j_call</th>\n",
       "      <th>decombinator_id</th>\n",
       "      <th>ID</th>\n",
       "      <th>control</th>\n",
       "      <th>PCR</th>\n",
       "      <th>chain</th>\n",
       "      <th>proportion_-3</th>\n",
       "      <th>...</th>\n",
       "      <th>total_-1</th>\n",
       "      <th>total_0</th>\n",
       "      <th>total_1</th>\n",
       "      <th>total_2</th>\n",
       "      <th>total_3</th>\n",
       "      <th>total_4</th>\n",
       "      <th>total_14</th>\n",
       "      <th>max_timepoint</th>\n",
       "      <th>max_timepoint_class</th>\n",
       "      <th>pgen</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>CASETSGSRLTF</td>\n",
       "      <td>TGTGCTTCTGAAACCAGTGGCTCTAGGTTGACCTTT</td>\n",
       "      <td>TRAV12-3</td>\n",
       "      <td>TRAJ58</td>\n",
       "      <td>5, 57, 8, 5, TTCT</td>\n",
       "      <td>17</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>alpha</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>85012.0</td>\n",
       "      <td>90223.0</td>\n",
       "      <td>78873.0</td>\n",
       "      <td>65517.0</td>\n",
       "      <td>65456.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>proportion_0</td>\n",
       "      <td>und</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>CAVRSARQLTF</td>\n",
       "      <td>TGTGCTGTTCGTTCTGCAAGGCAACTGACCTTT</td>\n",
       "      <td>TRAV8-2</td>\n",
       "      <td>TRAJ22</td>\n",
       "      <td>39, 11, 10, 9, CTGTTC</td>\n",
       "      <td>17</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>alpha</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>85012.0</td>\n",
       "      <td>90223.0</td>\n",
       "      <td>78873.0</td>\n",
       "      <td>65517.0</td>\n",
       "      <td>65456.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>proportion_2</td>\n",
       "      <td>und</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>CQPLGQNFVF</td>\n",
       "      <td>TGTCAACCCTTGGGTCAGAATTTTGTCTTT</td>\n",
       "      <td>TRAV3</td>\n",
       "      <td>TRAJ26</td>\n",
       "      <td>24, 14, 13, 11, CAACCCTTG</td>\n",
       "      <td>35</td>\n",
       "      <td>False</td>\n",
       "      <td>2.0</td>\n",
       "      <td>alpha</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>34950.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>73037.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>und</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>CVVSDQTGANNLFF</td>\n",
       "      <td>TGTGTTGTGAGTGATCAAACTGGGGCAAACAACCTCTTCTTT</td>\n",
       "      <td>TRAV8-2</td>\n",
       "      <td>TRAJ36</td>\n",
       "      <td>39, 24, 0, 0,</td>\n",
       "      <td>35</td>\n",
       "      <td>False</td>\n",
       "      <td>2.0</td>\n",
       "      <td>alpha</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>34950.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>73037.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>und</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>CVVSVRGATADMRF</td>\n",
       "      <td>TGTGTTGTGAGTGTACGGGGGGCCACGGCTGACATGCGCTTT</td>\n",
       "      <td>TRAV8-2</td>\n",
       "      <td>TRAJ43</td>\n",
       "      <td>39, 32, 1, 10, TACGGGGGGCCACGGC</td>\n",
       "      <td>35</td>\n",
       "      <td>False</td>\n",
       "      <td>2.0</td>\n",
       "      <td>alpha</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>34950.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>73037.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>und</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3900</th>\n",
       "      <td>CAVPDILTGGGNKLTF</td>\n",
       "      <td>TGTGCTGTTCCCGATATACTCACGGGAGGAGGAAACAAACTCACCTTT</td>\n",
       "      <td>TRAV3</td>\n",
       "      <td>TRAJ10</td>\n",
       "      <td>24, 0, 8, 0, TCCCGAT</td>\n",
       "      <td>371</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>alpha</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>91999.0</td>\n",
       "      <td>60696.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>71521.0</td>\n",
       "      <td>83993.0</td>\n",
       "      <td>44870.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>und</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3921</th>\n",
       "      <td>CACGVLHC</td>\n",
       "      <td>TGTGCGTGCGGTGTGCTGCATTGC</td>\n",
       "      <td>TRAV13-2</td>\n",
       "      <td>TRAJ35</td>\n",
       "      <td>7, 54, 8, 15, GTGCGG</td>\n",
       "      <td>373</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>alpha</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>160288.0</td>\n",
       "      <td>98933.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>159120.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>98300.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>und</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3983</th>\n",
       "      <td>CAPERGVSF</td>\n",
       "      <td>TGTGCCCCTGAAAGGGGGGTCAGCTTT</td>\n",
       "      <td>TRAV2</td>\n",
       "      <td>TRAJ20</td>\n",
       "      <td>13, 9, 9, 18, CCCTGAAAGGGGGG</td>\n",
       "      <td>373</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>alpha</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>160288.0</td>\n",
       "      <td>98933.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>159120.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>98300.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>und</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4028</th>\n",
       "      <td>CAVSGSNYGQNFVF</td>\n",
       "      <td>TGTGCTGTGAGTGGGTCCAACTATGGTCAGAATTTTGTCTTT</td>\n",
       "      <td>TRAV8-2</td>\n",
       "      <td>TRAJ26</td>\n",
       "      <td>39, 14, 10, 5, CTGTGAGTGGGTCC</td>\n",
       "      <td>373</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>alpha</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>160288.0</td>\n",
       "      <td>98933.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>159120.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>98300.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>und</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4032</th>\n",
       "      <td>CAVYGTSGSRLTF</td>\n",
       "      <td>TGTGCTGTCTACGGAACCAGTGGCTCTAGGTTGACCTTT</td>\n",
       "      <td>TRAV41</td>\n",
       "      <td>TRAJ58</td>\n",
       "      <td>34, 57, 3, 7, TACGG</td>\n",
       "      <td>373</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>alpha</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>160288.0</td>\n",
       "      <td>98933.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>159120.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>98300.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>und</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>114 rows Ã— 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           junction_aa                                          junction  \\\n",
       "24        CASETSGSRLTF              TGTGCTTCTGAAACCAGTGGCTCTAGGTTGACCTTT   \n",
       "55         CAVRSARQLTF                 TGTGCTGTTCGTTCTGCAAGGCAACTGACCTTT   \n",
       "68          CQPLGQNFVF                    TGTCAACCCTTGGGTCAGAATTTTGTCTTT   \n",
       "70      CVVSDQTGANNLFF        TGTGTTGTGAGTGATCAAACTGGGGCAAACAACCTCTTCTTT   \n",
       "74      CVVSVRGATADMRF        TGTGTTGTGAGTGTACGGGGGGCCACGGCTGACATGCGCTTT   \n",
       "...                ...                                               ...   \n",
       "3900  CAVPDILTGGGNKLTF  TGTGCTGTTCCCGATATACTCACGGGAGGAGGAAACAAACTCACCTTT   \n",
       "3921          CACGVLHC                          TGTGCGTGCGGTGTGCTGCATTGC   \n",
       "3983         CAPERGVSF                       TGTGCCCCTGAAAGGGGGGTCAGCTTT   \n",
       "4028    CAVSGSNYGQNFVF        TGTGCTGTGAGTGGGTCCAACTATGGTCAGAATTTTGTCTTT   \n",
       "4032     CAVYGTSGSRLTF           TGTGCTGTCTACGGAACCAGTGGCTCTAGGTTGACCTTT   \n",
       "\n",
       "        v_call  j_call                  decombinator_id   ID  control  PCR  \\\n",
       "24    TRAV12-3  TRAJ58                5, 57, 8, 5, TTCT   17     True  NaN   \n",
       "55     TRAV8-2  TRAJ22            39, 11, 10, 9, CTGTTC   17     True  NaN   \n",
       "68       TRAV3  TRAJ26        24, 14, 13, 11, CAACCCTTG   35    False  2.0   \n",
       "70     TRAV8-2  TRAJ36                   39, 24, 0, 0,    35    False  2.0   \n",
       "74     TRAV8-2  TRAJ43  39, 32, 1, 10, TACGGGGGGCCACGGC   35    False  2.0   \n",
       "...        ...     ...                              ...  ...      ...  ...   \n",
       "3900     TRAV3  TRAJ10             24, 0, 8, 0, TCCCGAT  371    False  0.0   \n",
       "3921  TRAV13-2  TRAJ35             7, 54, 8, 15, GTGCGG  373    False  0.0   \n",
       "3983     TRAV2  TRAJ20     13, 9, 9, 18, CCCTGAAAGGGGGG  373    False  0.0   \n",
       "4028   TRAV8-2  TRAJ26    39, 14, 10, 5, CTGTGAGTGGGTCC  373    False  0.0   \n",
       "4032    TRAV41  TRAJ58              34, 57, 3, 7, TACGG  373    False  0.0   \n",
       "\n",
       "      chain  proportion_-3  ...  total_-1   total_0  total_1  total_2  \\\n",
       "24    alpha            NaN  ...       NaN   85012.0  90223.0  78873.0   \n",
       "55    alpha            NaN  ...       NaN   85012.0  90223.0  78873.0   \n",
       "68    alpha            NaN  ...       NaN       NaN      NaN  34950.0   \n",
       "70    alpha            NaN  ...       NaN       NaN      NaN  34950.0   \n",
       "74    alpha            NaN  ...       NaN       NaN      NaN  34950.0   \n",
       "...     ...            ...  ...       ...       ...      ...      ...   \n",
       "3900  alpha            NaN  ...       NaN   91999.0  60696.0      NaN   \n",
       "3921  alpha            NaN  ...       NaN  160288.0  98933.0      NaN   \n",
       "3983  alpha            NaN  ...       NaN  160288.0  98933.0      NaN   \n",
       "4028  alpha            NaN  ...       NaN  160288.0  98933.0      NaN   \n",
       "4032  alpha            NaN  ...       NaN  160288.0  98933.0      NaN   \n",
       "\n",
       "       total_3  total_4  total_14  max_timepoint  max_timepoint_class  pgen  \n",
       "24     65517.0  65456.0       NaN   proportion_0                  und   0.0  \n",
       "55     65517.0  65456.0       NaN   proportion_2                  und   0.0  \n",
       "68         NaN      NaN   73037.0            NaN                  und   0.0  \n",
       "70         NaN      NaN   73037.0            NaN                  und   0.0  \n",
       "74         NaN      NaN   73037.0            NaN                  und   0.0  \n",
       "...        ...      ...       ...            ...                  ...   ...  \n",
       "3900   71521.0  83993.0   44870.0            NaN                  und   0.0  \n",
       "3921  159120.0      NaN   98300.0            NaN                  und   0.0  \n",
       "3983  159120.0      NaN   98300.0            NaN                  und   0.0  \n",
       "4028  159120.0      NaN   98300.0            NaN                  und   0.0  \n",
       "4032  159120.0      NaN   98300.0            NaN                  und   0.0  \n",
       "\n",
       "[114 rows x 30 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfa.loc[dfa['pgen'] == 0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "dfa.to_csv('../data/output_data/exp_A_wide5_pgen.csv')\n",
    "dfb.to_csv('../data/output_data/exp_B_wide5_pgen.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "     max_timepoint_class  chain variable         value   pgen_log\n",
      "0                    und  alpha     pgen  1.684988e-09  -8.773403\n",
      "1                    und  alpha     pgen  1.796038e-07  -6.745685\n",
      "2                    und  alpha     pgen  9.067683e-06  -5.042504\n",
      "3                    und  alpha     pgen  3.348758e-08  -7.475116\n",
      "4                    und  alpha     pgen  4.041352e-10  -9.393473\n",
      "...                  ...    ...      ...           ...        ...\n",
      "4070                 und  alpha     pgen  1.259741e-08  -7.899719\n",
      "4071                 und  alpha     pgen  8.216401e-07  -6.085318\n",
      "4072                 und  alpha     pgen  6.149321e-08  -7.211173\n",
      "4073                 und  alpha     pgen  1.154439e-07  -6.937629\n",
      "4074                 und  alpha     pgen  2.864361e-11 -10.542972\n",
      "\n",
      "[3961 rows x 5 columns]\n",
      "     max_timepoint_class chain variable         value   pgen_log\n",
      "0                    und  beta     pgen  1.122590e-10  -9.949779\n",
      "1                    und  beta     pgen  7.886121e-14 -13.103137\n",
      "2                    und  beta     pgen  2.046607e-09  -8.688965\n",
      "3                    und  beta     pgen  4.164285e-13 -12.380460\n",
      "4                    und  beta     pgen  5.674456e-10  -9.246076\n",
      "...                  ...   ...      ...           ...        ...\n",
      "5453                 und  beta     pgen  7.008734e-10  -9.154360\n",
      "5454                 und  beta     pgen  7.737058e-10  -9.111424\n",
      "5455                 und  beta     pgen  1.841960e-10  -9.734720\n",
      "5456                 und  beta     pgen  2.932297e-13 -12.532792\n",
      "5457                 und  beta     pgen  1.934386e-14 -13.713457\n",
      "\n",
      "[5448 rows x 5 columns]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/marti/miniconda3/envs/mm/lib/python3.10/site-packages/pandas/core/arraylike.py:402: RuntimeWarning: divide by zero encountered in log10\n",
      "  result = getattr(ufunc, method)(*inputs, **kwargs)\n",
      "/home/marti/miniconda3/envs/mm/lib/python3.10/site-packages/pandas/core/arraylike.py:402: RuntimeWarning: divide by zero encountered in log10\n",
      "  result = getattr(ufunc, method)(*inputs, **kwargs)\n"
     ]
    }
   ],
   "source": [
    "dfa_melt = pd.melt(dfa[['pgen', 'max_timepoint_class', 'chain']], id_vars = ['max_timepoint_class', 'chain'])\n",
    "dfa_melt['pgen_log'] = np.log10(dfa_melt['value'])\n",
    "dfa_melt = dfa_melt.loc[dfa_melt['value'] > 0]\n",
    "print(dfa_melt)\n",
    "\n",
    "dfb_melt = pd.melt(dfb[['pgen', 'max_timepoint_class', 'chain']], id_vars = ['max_timepoint_class', 'chain'])\n",
    "dfb_melt['pgen_log'] = np.log10(dfb_melt['value'])\n",
    "dfb_melt = dfb_melt.loc[dfb_melt['value'] > 0]\n",
    "print(dfb_melt)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "# colormap = {'late':'cyan', 'early':'orange', 'und':'lightgrey'}\n",
    "# colormap_edge = {'late':'darkcyan', 'early':'darkorange', 'und':'darkgrey'}\n",
    "\n",
    "# ax = plt.subplot()\n",
    "# sns.kdeplot(data=dfa_melt.loc[dfa_melt['max_timepoint_class'] == 'und'], \n",
    "#             x = 'pgen_log', ax=ax, label = 'und', color=colormap['und'])\n",
    "# for c in ['early', 'late']:\n",
    "#     print(c)\n",
    "#     sns.histplot(data=dfa_melt.loc[dfa_melt['max_timepoint_class'] == c], \n",
    "#                 x = 'pgen_log', ax=ax, label = c, color=colormap[c], edgecolor =colormap_edge[c],\n",
    "#                 binwidth=1, binrange=(-20,-4), stat='proportion', alpha = 0.3)\n",
    "# plt.legend()\n",
    "# plt.xlim(-20, -4)\n",
    "# plt.xlabel(r'log$_{10}$(pgen)')\n",
    "# plt.ylabel('Proportion of sample')\n",
    "# plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "# colormap = {'late':'cyan', 'early':'orange', 'und':'lightgrey'}\n",
    "# colormap_edge = {'late':'darkcyan', 'early':'darkorange', 'und':'darkgrey'}\n",
    "\n",
    "# ax = plt.subplot()\n",
    "# sns.kdeplot(data=dfb_melt.loc[dfb_melt['max_timepoint_class'] == 'und'], \n",
    "#             x = 'pgen_log', ax=ax, label = 'und', color=colormap['und'])\n",
    "# for c in ['early', 'late']:\n",
    "#     print(c)\n",
    "#     sns.histplot(data=dfb_melt.loc[dfb_melt['max_timepoint_class'] == c], \n",
    "#                 x = 'pgen_log', ax=ax, label = c, color=colormap[c], edgecolor =colormap_edge[c],\n",
    "#                 binwidth=1, binrange=(-16,-4), stat='proportion', alpha = 0.3)\n",
    "# plt.legend()\n",
    "# plt.xlim(-20, -4)\n",
    "# plt.xlabel(r'log$_{10}$(pgen)')\n",
    "# plt.ylabel('Proportion of sample')\n",
    "# plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "LCMV = pd.DataFrame()\n",
    "\n",
    "for f in os.listdir('../data/LCMVdata/'):\n",
    "    if ('E_' in f) and ('_freq' in f):\n",
    "        lcmv = pd.read_csv('../data/LCMVdata/' + f, index_col=0).reset_index(names='junction_aa')\n",
    "        lcmv = pd.melt(lcmv, id_vars = ['junction_aa'])\n",
    "        lcmv = lcmv.loc[lcmv['value'] > 0]\n",
    "        lcmv[['epitope', 'condition', 'mouse']] = lcmv['variable'].str.split(' ', expand=True)\n",
    "        LCMV = pd.concat([LCMV, lcmv]).reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>junction_aa</th>\n",
       "      <th>variable</th>\n",
       "      <th>value</th>\n",
       "      <th>epitope</th>\n",
       "      <th>condition</th>\n",
       "      <th>mouse</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CAAGTGDSDYTF</td>\n",
       "      <td>NP205 LCMV40 M1</td>\n",
       "      <td>0.000076</td>\n",
       "      <td>NP205</td>\n",
       "      <td>LCMV40</td>\n",
       "      <td>M1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CACRDRGNQNTLYF</td>\n",
       "      <td>NP205 LCMV40 M1</td>\n",
       "      <td>0.000312</td>\n",
       "      <td>NP205</td>\n",
       "      <td>LCMV40</td>\n",
       "      <td>M1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CADRGVQNTLYF</td>\n",
       "      <td>NP205 LCMV40 M1</td>\n",
       "      <td>0.002006</td>\n",
       "      <td>NP205</td>\n",
       "      <td>LCMV40</td>\n",
       "      <td>M1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CASADRGQNTLYF</td>\n",
       "      <td>NP205 LCMV40 M1</td>\n",
       "      <td>0.000151</td>\n",
       "      <td>NP205</td>\n",
       "      <td>LCMV40</td>\n",
       "      <td>M1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CASADRYNSPLYF</td>\n",
       "      <td>NP205 LCMV40 M1</td>\n",
       "      <td>0.000303</td>\n",
       "      <td>NP205</td>\n",
       "      <td>LCMV40</td>\n",
       "      <td>M1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      junction_aa         variable     value epitope condition mouse\n",
       "0    CAAGTGDSDYTF  NP205 LCMV40 M1  0.000076   NP205    LCMV40    M1\n",
       "1  CACRDRGNQNTLYF  NP205 LCMV40 M1  0.000312   NP205    LCMV40    M1\n",
       "2    CADRGVQNTLYF  NP205 LCMV40 M1  0.002006   NP205    LCMV40    M1\n",
       "3   CASADRGQNTLYF  NP205 LCMV40 M1  0.000151   NP205    LCMV40    M1\n",
       "4   CASADRYNSPLYF  NP205 LCMV40 M1  0.000303   NP205    LCMV40    M1"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "LCMV.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "generative_model = load_model.GenerativeModelVDJ()\n",
    "marginals_file_name = '../../OLGA/olga/default_models/mouse_T_beta/model_marginals.txt'\n",
    "generative_model.load_and_process_igor_model(marginals_file_name)\n",
    "\n",
    "params_file_name = '../../OLGA/olga/default_models/mouse_T_beta/model_params.txt'\n",
    "V_anchor_pos_file ='../../OLGA/olga/default_models/mouse_T_beta/V_gene_CDR3_anchors.csv'\n",
    "J_anchor_pos_file = '../../OLGA/olga/default_models/mouse_T_beta/J_gene_CDR3_anchors.csv'\n",
    "genomic_data = load_model.GenomicDataVDJ()\n",
    "genomic_data.load_igor_genomic_data(params_file_name, V_anchor_pos_file, J_anchor_pos_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "pgen_b_mouse = pgen.GenerationProbabilityVDJ(generative_model, genomic_data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "LCMV['pgen'] = LCMV.apply(lambda row: \n",
    "                        pgen_b_mouse.compute_aa_CDR3_pgen(row['junction_aa'], \n",
    "                        print_warnings = True), axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "LCMV.to_csv('../data/output_data/LCMV_pgen.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>junction_aa</th>\n",
       "      <th>variable</th>\n",
       "      <th>value</th>\n",
       "      <th>epitope</th>\n",
       "      <th>condition</th>\n",
       "      <th>mouse</th>\n",
       "      <th>pgen</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CAAGTGDSDYTF</td>\n",
       "      <td>NP205 LCMV40 M1</td>\n",
       "      <td>0.000076</td>\n",
       "      <td>NP205</td>\n",
       "      <td>LCMV40</td>\n",
       "      <td>M1</td>\n",
       "      <td>2.830681e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CACRDRGNQNTLYF</td>\n",
       "      <td>NP205 LCMV40 M1</td>\n",
       "      <td>0.000312</td>\n",
       "      <td>NP205</td>\n",
       "      <td>LCMV40</td>\n",
       "      <td>M1</td>\n",
       "      <td>3.241128e-11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CADRGVQNTLYF</td>\n",
       "      <td>NP205 LCMV40 M1</td>\n",
       "      <td>0.002006</td>\n",
       "      <td>NP205</td>\n",
       "      <td>LCMV40</td>\n",
       "      <td>M1</td>\n",
       "      <td>2.133456e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CASADRGQNTLYF</td>\n",
       "      <td>NP205 LCMV40 M1</td>\n",
       "      <td>0.000151</td>\n",
       "      <td>NP205</td>\n",
       "      <td>LCMV40</td>\n",
       "      <td>M1</td>\n",
       "      <td>1.348938e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CASADRYNSPLYF</td>\n",
       "      <td>NP205 LCMV40 M1</td>\n",
       "      <td>0.000303</td>\n",
       "      <td>NP205</td>\n",
       "      <td>LCMV40</td>\n",
       "      <td>M1</td>\n",
       "      <td>5.485257e-17</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      junction_aa         variable     value epitope condition mouse  \\\n",
       "0    CAAGTGDSDYTF  NP205 LCMV40 M1  0.000076   NP205    LCMV40    M1   \n",
       "1  CACRDRGNQNTLYF  NP205 LCMV40 M1  0.000312   NP205    LCMV40    M1   \n",
       "2    CADRGVQNTLYF  NP205 LCMV40 M1  0.002006   NP205    LCMV40    M1   \n",
       "3   CASADRGQNTLYF  NP205 LCMV40 M1  0.000151   NP205    LCMV40    M1   \n",
       "4   CASADRYNSPLYF  NP205 LCMV40 M1  0.000303   NP205    LCMV40    M1   \n",
       "\n",
       "           pgen  \n",
       "0  2.830681e-08  \n",
       "1  3.241128e-11  \n",
       "2  2.133456e-09  \n",
       "3  1.348938e-07  \n",
       "4  5.485257e-17  "
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "LCMV.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "     condition epitope variable         value   pgen_log\n",
      "0       LCMV40   NP205     pgen  2.830681e-08  -7.548109\n",
      "1       LCMV40   NP205     pgen  3.241128e-11 -10.489304\n",
      "2       LCMV40   NP205     pgen  2.133456e-09  -8.670916\n",
      "3       LCMV40   NP205     pgen  1.348938e-07  -6.870008\n",
      "4       LCMV40   NP205     pgen  5.485257e-17 -16.260803\n",
      "...        ...     ...      ...           ...        ...\n",
      "2330       PBS    GP92     pgen  1.928567e-06  -5.714765\n",
      "2331       PBS    GP92     pgen  3.475092e-06  -5.459034\n",
      "2332       PBS    GP92     pgen  1.546273e-05  -4.810714\n",
      "2333       PBS    GP92     pgen  1.004486e-06  -5.998056\n",
      "2334       PBS    GP92     pgen  3.860320e-10  -9.413377\n",
      "\n",
      "[2335 rows x 5 columns]\n"
     ]
    }
   ],
   "source": [
    "LCMV_melt = pd.melt(LCMV[['pgen', 'condition', 'epitope']], id_vars = ['condition', 'epitope'])\n",
    "LCMV_melt['pgen_log'] = np.log10(LCMV_melt['value'])\n",
    "LCMV_melt = LCMV_melt.loc[LCMV_melt['value'] > 0]\n",
    "print(LCMV_melt)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "# colormap = {'LCMV40':'cyan', 'LCMV8':'orange', 'PBS':'lightgrey'}\n",
    "# colormap_edge = {'LCMV40':'darkcyan', 'LCMV8':'darkorange', 'PBS':'darkgrey'}\n",
    "\n",
    "# ax = plt.subplot()\n",
    "# sns.kdeplot(data=LCMV_melt.loc[LCMV_melt['condition'] == 'PBS'], \n",
    "#             x = 'pgen_log', ax=ax, label = 'PBS', color=colormap['PBS'])\n",
    "# for c in ['LCMV40', 'LCMV8']:\n",
    "#     print(c)\n",
    "#     sns.histplot(data=LCMV_melt.loc[LCMV_melt['condition'] == c], \n",
    "#                 x = 'pgen_log', ax=ax, label = c, color=colormap[c], edgecolor =colormap_edge[c],\n",
    "#                 binwidth=1, binrange=(-20,-3), stat='proportion', alpha = 0.3)\n",
    "#     # sns.kdeplot(data=LCMV_melt.loc[LCMV_melt['condition'] == c], \n",
    "#     #             x = 'pgen_log', ax=ax, label = c, color=colormap[c])\n",
    "# plt.legend()\n",
    "# plt.xlim(-20, -3)\n",
    "# plt.xlabel(r'log$_{10}$(pgen)')\n",
    "# plt.ylabel('Proportion of sample')\n",
    "# plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "olga",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
